### ISSUE TITLE
Add keyframe-block-no-duplicate-selectors

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#5991]
### What is the problem you're trying to solve?

When revisiting https://github.com/stylelint/stylelint/issues/4065, I noticed on the MDN site that [duplicate selectors in keyframe-blocks are are thing](https://developer.mozilla.org/en-US/docs/Web/CSS/@keyframes#when_a_keyframe_is_defined_multiple_times).

Also show in example 3 [in the spec](https://drafts.csswg.org/css-animations/#keyframes).

I'd like to disallow them as, even though it's valid CSS, I've probably made a mistake.

### What solution would you like to see?

A new rule like the existing [declaration-block-no-duplicate-properties](https://stylelint.io/user-guide/rules/list/declaration-block-no-duplicate-properties) one.

- Name: `keyframe-block-no-duplicate-selectors`
- Primary option: `true`
- Secondary options: None
- Autofixable: No
- Message: `Unexpected duplicate "${selector}"`
- Description: "Disallow duplicate selectors within keyframe blocks."
- Section: "Avoid errors" -> "Keyframe blocks"

[Language in the spec](https://drafts.csswg.org/css-animations/#keyframes) for `keyframe-block` and `selector`.

It would disallow the likes of:

```css
@keyframes foo {
  0% {}
  0% {}
  100% {}
}
```

Sound good?

**Comment by jeddy3** [comment]
As there are no objections, I'll label it as good to go.

**Comment by ybiquitous** [comment]
I like this rule. 👍🏼

**Comment by mattxwang** [comment]
I'd be willing to take this on!

To quickly write out my approach (which could certainly be wrong):

1. I see that I can get each `@keyframe` with `walkAtRules(/keyframes/i, ...)`
2. Then, in each `@keyframe`, I'll initialize a set and then walk each selector; on a duplicate selector, I'll error

Is that the most efficient way to do this?

**Comment by jeddy3** [comment]
That sounds good to me, @mattxwang.

You can use the [`declaration-block-no-duplicate-custom-properties`](https://github.com/stylelint/stylelint/blob/main/lib/rules/declaration-block-no-duplicate-custom-properties/index.js) rule as a blueprint as its more stripped back than the `*-properties` one.
